sortedcontainers/sortedlist.pyi

```diff
@@ -0,0 +1,170 @@
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+)
+
+_T = TypeVar("_T")
+_SL = TypeVar("_SL", bound="SortedList")
+_SKL = TypeVar("_SKL", bound="SortedKeyList")
+_Key = Callable[[_T], Any]
+_Repr = Callable[[], str]
+
+def recursive_repr(fillvalue: str = ...) -> Callable[[_Repr], _Repr]: ...
+
+
+class SortedList(MutableSequence[_T]):
+
+    DEFAULT_LOAD_FACTOR: int = ...
+    def __init__(self, iterable: Optional[Iterable[_T]] = ..., key: Optional[_Key] = ...): ...
+
+    # NB: currently mypy does not honour return type, see mypy #3307
+    @typing.overload
+    def __new__(cls: Type[_SL], iterable: None, key: None) -> _SL: ...
+    @typing.overload
+    def __new__(cls: Type[_SL], iterable: None, key: _Key) -> SortedKeyList[_T]: ...
+    @typing.overload
+    def __new__(cls: Type[_SL], iterable: Iterable[_T], key: None) -> _SL: ...
+    @typing.overload
+    def __new__(cls, iterable: Iterable[_T], key: _Key) -> SortedKeyList[_T]: ...
+
+    @property
+    def key(self) -> Optional[Callable[[_T], Any]]: ...
+    def _reset(self, load: int) -> None: ...
+    def clear(self) -> None: ...
+    def _clear(self) -> None: ...
+    def add(self, value: _T) -> None: ...
+    def _expand(self, pos: int) -> None: ...
+    def update(self, iterable: Iterable[_T]) -> None: ...
+    def _update(self, iterable: Iterable[_T]) -> None:...
+    def discard(self, value: _T) -> None: ...
+    def remove(self, value: _T) -> None: ...
+    def _delete(self, pos: int, idx: int) -> None: ...
+    def _loc(self, pos: int, idx: int) -> int: ...
+    def _pos(self, idx: int) -> int: ...
+    def _build_index(self) -> None: ...
+    def __contains__(self, value: Any) -> bool: ...
+    def __delitem__(self, index: Union[int, slice]) -> None: ...
+    @typing.overload
+    def __getitem__(self, index: int) -> _T: ...
+    @typing.overload
+    def __getitem__(self, index: slice) -> List[_T]: ...
+    @typing.overload
+    def _getitem(self, index: int) -> _T: ...
+    @typing.overload
+    def _getitem(self, index: slice) -> List[_T]: ...
+    @typing.overload
+    def __setitem__(self, index: int, value: _T) -> None: ...
+    @typing.overload
+    def __setitem__(self, index: slice, value: Iterable[_T]) -> None: ...
+    def __iter__(self) -> Iterator[_T]: ...
+    def __reversed__(self) -> Iterator[_T]: ...
+    def __len__(self) -> int: ...
+    def reverse(self) -> None: ...
+    def islice(
+        self, start: Optional[int] = ..., stop: Optional[int] = ..., reverse=bool
+    ) -> Iterator[_T]: ...
+    def _islice(
+        self, min_pos: int, min_idx: int, max_pos: int, max_idx: int, reverse: bool
+    ) -> Iterator[_T]: ...
+    def irange(
+        self,
+        minimum: Optional[int] = ...,
+        maximum: Optional[int] = ...,
+        inclusive: Tuple[bool, bool] = ...,
+        reverse: bool = ...,
+    ) -> Iterator[_T]: ...
+    def bisect_left(self, value: _T) -> int: ...
+    def bisect_right(self, value: _T) -> int: ...
+    bisect = bisect_right
```

Comment:
This and all the other assigned methods in this file should be inlined like what was done in `SortedDict`